### PR TITLE
Allows to skip checks (all or default)

### PR DESCRIPTION
**Pull request by Flink**
Setting `DOKKU_SKIP_ALL_CHECKS` globally or for a given app will skip
all checks.
Setting `DOKKU_SKIP_DEFAULT_CHECKS` globally or for a given app will skip
default checks only.

### CODE REVIEW

**Comment by alessio** [comment]
:+1:

File: dokku [hunk 1]
```diff
@@ -70,6 +70,9 @@ case "$1" in
     DOKKU_SCALE_FILE="$DOKKU_ROOT/$APP/DOKKU_SCALE"
     oldids=$(get_container_ids $APP)
 
+    [[ -f "$DOKKU_ROOT/ENV" ]] && source $DOKKU_ROOT/ENV
+    [[ -f "$DOKKU_ROOT/$APP/ENV" ]] && source $DOKKU_ROOT/$APP/ENV
```
Comment (michaelshobbs) [inline dokku]: these lines may have wider consequences than intended. we've seen in the past where we've added sourcing of app envs cause some issues. that isn't to say these line should be changed but that we may want to remember them specifically when troubleshooting after this gets merged. cc: @josegonzalez

**Comment by michaelshobbs** [comment]
:+1:

File: dokku (hunk 1 above)
Comment (josegonzalez) [inline dokku]: Why not just retrieve the value via config:get ?

Comment (Flink) [inline dokku]: Because if I’m not mistaken, you can’t get global config with `config:get`

**Comment by michaelshobbs** [comment]
@josegonzalez @Flink anything else to do here or shall I merge?

**Comment by Flink** [comment]
@michaelshobbs it’s good for me, I don’t know if @josegonzalez saw my answer the other day ;)

File: dokku (hunk 1 above)
Comment (josegonzalez) [inline dokku]: Can you do this some other way? I'm mostly afraid someone will define something like `PROC_TYPE` and break all the things.

Perhaps it's time to have a `globalconfig:get` method?

Comment (Flink) [inline dokku]: I was thinking we could have a special app named `dokku` to set and get global config?

**Comment by michaelshobbs** [comment]
I'm :-1: on a special app name. I thinking keeping this stuff in the config plugin makes sense. Perhaps just `config:globalset` or `config:set --global`. I think I like the UX of the second more even though I'm not a fan of the additional arg parsing but whatevs.

**Comment by Flink** [comment]
The `--global` flag seems nice but I think it lacks consistency with the existing UX. If we use it, we would have to check that an application name is not given.
Maybe we could use `config:set-global` without an app name, since there is already a `config:set-norestart`, the pattern will remain the same.
What do you think?

**Comment by josegonzalez** [comment]
@Flink +1

**Comment by michaelshobbs** [comment]
@Flink yeah my point there was to make it a bit inconsistent because it's not setting and app's config. If we parse for `--` args up front then we just bypass the whole $APP check anyway.

Either way, looking at the code now, it'll be a significant (but probably necessary) change to the config plugin.

**Comment by Flink** [comment]
So do I add a `config:set-global` action or not? :)

**Comment by michaelshobbs** [comment]
I think we should overload `config:set` and `config:get` with `--global`. I'd like to hear @josegonzalez weigh in with his thoughts (more than a :+1:) :smile:

**Comment by Flink** [comment]
Ok then :wink:

**Comment by josegonzalez** [comment]
@michaelshobbs I'd rather what @Flink proposed, assuming we're going to keep `config:set-norestart`. Seems like we should standardize on one method of `config` setting/getting. Though I'm happy if the flag-based method is implemented, and we deprecate `config:set-norestart` :)

**Comment by michaelshobbs** [comment]
@josegonzalez Upon much reflection after implementing the `config:set-norestart` syntax, I think I'd prefer the flag-based method. @Flink let me know if you want to run with that. otherwise, i can.

**Comment by josegonzalez** [comment]
@michaelshobbs makes sense to me, just want to expose standard interfaces where possible. If we have a new standard, good :).

**Comment by Flink** [comment]
@michaelshobbs ok for the new syntax so ;) I think you should give it a try since it seems you already have something in mind about implementation :grin:

**Comment by michaelshobbs** [comment]
@Flink we can also work together on it. i can submit a PR against your branch if you'd like.

**Comment by josegonzalez** [comment]
@michaelshobbs Maybe you work on your pr, merge to master, and @Flink can rebase?

**Comment by Flink** [comment]
Yes I agree with @josegonzalez, it seems more natural to me this way. But thanks @michaelshobbs :)

**Comment by michaelshobbs** [comment]
sounds good

**Comment by josegonzalez** [comment]
Also makes it easy for me to make release notes :)

**Comment by michaelshobbs** [comment]
`dokku config:get --global DOKKU_SKIP_ALL_CHECKS` will work now

**Comment by josegonzalez** [comment]
Okay, *now* `dokku config:get --global DOKKU_SKIP_ALL_CHECKS` will work.

**Comment by josegonzalez** [comment]
I think I'l going to merge this for now. @michaelshobbs Mind making a separate pr that updates all places where we retrieve global config manually?

**Pull request merged**